config/routes.rb

```diff
@@ -3,5 +3,6 @@
 
   namespace :v1 do
     resources :projects, only: crud
+    resources :tasks, only: %i(index show create)
```

Comment:
My thought was to keep clean endpoints with limited nesting. Querying an endpoint gets you resource related data and explicit query params filters the data.